pkgs/servers/bloat/default.nix

```
@@ -0,0 +1,43 @@
+{ lib
+, buildGoModule
+, fetchgit
+, unstableGitUpdater
+}:
+
+buildGoModule {
+  pname = "bloat";
+  version = "unstable-2022-02-11";
+
+  src = fetchgit {
+    url = "git://git.freesoftwareextremist.com/bloat";
```

```suggestion
    url = "https://git.freesoftwareextremist.com/bloat";
```
`git://` is considered unsafe.

---

Yeah but we have the hash, and that is dumb http, so eg. --depth does not work and the whole repo has to be fetched. I guess I could also fetch a tarball, but that'd break unstableGitUpdater :-/

I'm not sure what's the best option here

---

Why would the whole repo not have to be fetched? What is your use of `--depth` here?

---

> Why would the whole repo not have to be fetched?

I meant the whole history

> What is your use of --depth here?

unstableGitUpdater only needs HEAD, so it uses `--depth 1`, and I assume that fetchgit does the same

---

is writing at least disabled over `git://`. If so I'd go with it.

---

I mean pushing to it

---

I have no idea, but only the owners are going to use it for writing, not nixpkgs users